src/worker/clustering_media/clustering_media_worker.py

```diff
@@ -0,0 +1,172 @@
+                        if video_config.get("tsne"):
+                            embedding = vid_vec_rep_clip.run(video_path)
+                            video_embeddings.append({
+                                "embedding": next(embedding)["vid_vec"],
+                                "payload": file_id
+                            })
+                    else:
+                        embedding = vid_vec_rep_clip.run(video_path)
+                        video_embeddings.append({
+                            "embedding": next(embedding)["vid_vec"],
+                            "payload": file_id
+                        })
+
+                except Exception as e:
+                    print("Error in generating embeddings", e)
+                    # send failed report to report queue
+                    report = make_report_failed(media_type, "failed", file_id)
+                    feluda.queue.message(
+                    feluda.config.queue.parameters.queues[1]["name"], report
+                    )
+                    # requeue the media file
+                    ch.basic_ack(delivery_tag=method.delivery_tag)
+            else:
+                pass
+
+        clustering_results_audio = cluster_embeddings.run(input_data=audio_embeddings, n_clusters=audio_config.get("n_clusters"), modality='audio')
+        if "labels" in video_config:
+            clustering_results_video = video_classifications
+        else:
+            clustering_results_video = cluster_embeddings.run(input_data=video_embeddings, n_clusters=video_config.get("n_clusters"), modality='video')
+        clustering_results_json = {
+            "audio": clustering_results_audio,
+            "video": clustering_results_video
+        }
+        # dim_reduction_results_json = []
+        # if audio_config.get("tsne"):
+        #     dim_reduction_results_json.extend(dimension_reduction.perform_reduction(audio_embeddings))
+        # if video_config.get("tsne"):
+        #     dim_reduction_results_json.extend(dimension_reduction.perform_reduction(video_embeddings))
+        report = make_report_indexed(clustering_results_json, "indexed")
+        feluda.queue.message(
+            feluda.config.queue.parameters.queues[1]["name"], report
+        )
+        ch.basic_ack(delivery_tag=method.delivery_tag)
+
+    return worker
+
+feluda = None
+clustering_media_index_queue = None
+try:
+    # Init Feluda and load config
+    feluda = Feluda("worker/clustering_media/config.yml")
+    feluda.setup()
+    clustering_media_index_queue = feluda.config.queue.parameters.queues[0]["name"]
+    # setup Components
+    feluda.start_component(ComponentType.QUEUE)
+
+    # init all operators
+    audio_vec_embedding_clap.initialize(param={})
+    vid_vec_rep_clip.initialize(param={})
+    classify_video_zero_shot.initialize(param={})
+    cluster_embeddings.initialize(param={})
+    # dimension_reduction.setup_reduction(model_type='tsne', params={})
```

Comment:
@Chaithanya512 I have commented out the dimension reduction operator as it does not adhere to the Feluda's operator interface conventions and hence was throwing an error when initializing Feluda with a config file. Every operator must only have the `initialize` and `run` methods so some refactoring is needed on the `dimension_reduction` operator.
Keeping it aside, the worker is now working properly on clustering as tested with the payload writer.